stl/inc/vector

```diff
@@ -458,6 +484,136 @@ public:
     using difference_type = typename _Alty_traits::difference_type;
 
 private:
+#if defined(__SANITIZE_ADDRESS__) || defined(_ANNOTATE_VECTOR)
+    class _NODISCARD _Asan_annotator {
+    public:
+        _Asan_annotator(const _Asan_annotator&) = delete;
+        _Asan_annotator& operator=(const _Asan_annotator&) = delete;
+
+        explicit constexpr _Asan_annotator(vector* _Myvec_, size_type _Count_) noexcept
+            : _Myvec(_Myvec_), _Count(_Count_) {
+            _STL_INTERNAL_CHECK(_Myvec != nullptr);
+            auto& _My_data = _Myvec->_Mypair._Myval2;
+            _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Mylast, _My_data._Mylast + _Count);
+        }
+
+        _CONSTEXPR20 ~_Asan_annotator() {
+            if (!_Myvec) { // Operation succeeded, no modification to the shadow memory required.
+                return;
+            }
+
+            // Shrinks the shadow memory to the current size of the vector.
+            auto& _My_data = _Myvec->_Mypair._Myval2;
+            _Apply_annotation(_My_data._Myfirst, _My_data._Myend, _My_data._Mylast + _Count, _My_data._Mylast);

```

Comment:
`_Mylast` may have changed by the time the destructor runs, but `_Myfirst` and `_Myend` won't. We can fix this problem by storing the target size instead of the delta in the constructor.

We could fix this by storing the target size instead of the delta in the constructor,

Reply:
I believe that your analysis is wrong.

The destructor will only ever do something if an exception was thrown. 

In that case, we should already have cleaned up `_Mylast` to point at the where it was before the exception.

Reply:
I observe that `insert` on lines 984-994 doesn't meet this precondition. In both branches, `_Mylast` is updated between annotator construction destruction and in neither branch restored to its construction value if an exception is thrown.